### DISCUSSION TITLE
AC <-> TC SmartAI diff

### DISCUSSION THREAD

**Discussion by Helias** [azerothcore/azerothcore-wotlk#3950] [General]
I compared the TC 335a database with the AC database about the SmartAI.
I've found that TC has 1913 SmartAI that we do not have.

Here the full list: https://paste.ubuntu.com/p/3J6rKNPfP9/

I used the following SQL script to get them.
```SQL
SELECT entry, name, AIName FROM TC.creature_template WHERE AIName != '' AND entry NOT IN (
	SELECT entry FROM AC.creature_template WHERE AIName != ''
) LIMIT 20000;
```

This could be a good list of "missing SAI", feel free to port them on AC ;-)

@Kitzunu @Knindzagxg

**Comment by Kitzunu** [comment]
You also need to check if we have core scripts assigned to them

**Comment by Knindzagxg** [comment]
My guild mates from retail did full checking of classic zones and so far a lot of problems.
But during this and next monht i should be able to sort all of them xP

**Comment by LichKing255** [comment]
> I compared the TC 335a database with the AC database about the SmartAI.
> I've found that TC has 1913 SmartAI that we do not have.
> 
> Here the full list: https://paste.ubuntu.com/p/3J6rKNPfP9/
> 
> I used the following SQL script to get them.
> 
> ```sql
> SELECT entry, name, AIName FROM TC.creature_template WHERE AIName != '' AND entry NOT IN (
> 	SELECT entry FROM AC.creature_template WHERE AIName != ''
> ) LIMIT 20000;
> ```
> 
> This could be a good list of "missing SAI", feel free to port them on AC ;-)
> 
> @Kitzunu @Knindzagxg
> 
> Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/91431261-ac-tc-smartai-diff?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).

your query is not true. it's true :

```sql
SELECT entry, name, AIName FROM TC.creature_template WHERE AIName != '' AND entry NOT IN (
	SELECT entry FROM AC.creature_template WHERE AIName = '' AND (ScriptName = NULL OR ScriptName = '')
) LIMIT 20000;
```

**Comment by FrancescoBorzi** [comment]
this is more of a forum talk, moved
